mne/report/report.py

```diff
@@ -2437,7 +2437,8 @@ def save(self, fname=None, open_browser=True, overwrite=False,
         if open_browser and not is_hdf5 and not building_doc:
             webbrowser.open_new_tab('file://' + fname)
 
-        self.fname = fname
+        if self.fname is None:
```

Comment:
To fix the failing test, I think this either this needs to be `getattr(self, 'fname', None)` or you need to add `fname` to `_get_state_params`